### PR TITLE
[HTTP] Refactor unhanded ballerina exceptions

### PR DESCRIPTION
**Pull request by chamil321**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/http/src/main/java/org/ballerinalang/net/http/actions/httpclient/AbstractHTTPAction.java [hunk 1]
```diff
@@ -130,15 +131,17 @@ static void prepareOutboundRequest(Strand strand, String serviceUri, String path
             setOutboundReqHeaders(outboundRequest, port, host);
 
         } catch (MalformedURLException e) {
-            throw new BallerinaException("Malformed url specified. " + e.getMessage());
+            throw HttpUtil.createHttpError("Malformed url specified. " + e.getMessage(),
```
Comment (pubudu91) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/actions/httpclient/AbstractHTTPAction.java]: url -> URL

File: stdlib/http/src/main/java/org/ballerinalang/net/http/actions/httpclient/AbstractHTTPAction.java [hunk 2]
```diff
@@ -130,15 +131,17 @@ static void prepareOutboundRequest(Strand strand, String serviceUri, String path
...
+                                           HttpErrorType.GENERIC_CLIENT_ERROR);
         } catch (Exception e) {
-            throw new BallerinaException("Failed to prepare request. " + e.getMessage());
+            throw HttpUtil.createHttpError("Failed to prepare request. " + e.getMessage(),
```
Comment (pubudu91) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/actions/httpclient/AbstractHTTPAction.java]: Inconsistent use of case. The earlier errors start with lower case letters

File: stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/ResponseWriter.java [hunk 3]
```diff
@@ -103,7 +103,8 @@ private static void serializeMultiparts(String boundaryString, ObjectValue entit
                 HttpUtil.closeMessageOutputStream(messageOutputStream);
             }
         } catch (IOException ex) {
-            throw new BallerinaException("Error occurred while serializing byte channel content : ", ex);
+            throw BallerinaErrors.createError(
+                    "Error occurred while serializing byte channel content : " + ex.getMessage());
```
Comment (pubudu91) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/ResponseWriter.java]: This should've been additional details to the error. Not the reason itself.

File: stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/ResponseWriter.java [hunk 4]
```diff
@@ -125,7 +126,8 @@ static void serializeDataSource(Object outboundMessageSource, ObjectValue entity
                 HttpUtil.closeMessageOutputStream(messageOutputStream);
             }
         } catch (IOException ex) {
-            throw new BallerinaException("Error occurred while serializing message data source : ", ex);
+            throw BallerinaErrors.createError(
+                    "Error occurred while serializing message data source : " + ex.getMessage());
```
Comment (pubudu91) [inline stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/ResponseWriter.java]: Same comment as above here

**Comment by chamil321** [comment]
Addressed comments in https://github.com/ballerina-platform/ballerina-lang/pull/18692

**Pull request merged**